bindings/rust/build.rs

```diff
@@ -1,170 +1,58 @@
-
-        let mut cmd = Command::new("cmake");
-
-        // We don't use TARGET since we can't cross-build.
-        if env::consts::OS == "windows" {
-            // Windows
-            cmd.current_dir(&unicorn_dir)
-                .arg("-B")
-                .arg("build_rust")
-                .arg("-DBUILD_SHARED_LIBS=OFF")
-                .arg("-G")
-                .arg("Visual Studio 16 2019");
-
-            if profile == "debug" {
-                cmd.arg("-DCMAKE_BUILD_TYPE=Debug");
-            } else {
-                cmd.arg("-DCMAKE_BUILD_TYPE=Release");
-            }
-
-            cmd.output()
-                .expect("Fail to create build directory on Windows.");
-
-            let mut platform = "x64";
-            let mut conf = "Release";
-            if std::mem::size_of::<usize>() == 4 {
-                platform = "Win32";
-            }
-            if profile == "debug" {
-                conf = "Debug";
-            }
-
-            Command::new("msbuild")
-                .current_dir(&rust_build_path)
-                .arg("unicorn.sln")
-                .arg("-m")
-                .arg("-p:Platform=".to_owned() + platform)
-                .arg("-p:Configuration=".to_owned() + conf)
-                .output()
-                .expect("Fail to build unicorn on Win32.");
-            println!(
-                "cargo:rustc-link-search={}",
-                rust_build_path.join(conf).to_str().unwrap()
-            );
-        } else {
-            // Most Unix-like systems
-            let mut cmd = Command::new("cmake");
-            cmd.current_dir(&unicorn_dir)
-                .arg("-B")
-                .arg("build_rust")
-                .arg("-DBUILD_SHARED_LIBS=OFF");
-
-            if profile == "debug" {
-                cmd.arg("-DCMAKE_BUILD_TYPE=Debug");
-            } else {
-                cmd.arg("-DCMAKE_BUILD_TYPE=Release");
-            }
-
-            cmd.output()
-                .expect("Fail to create build directory on *nix.");
-
-            Command::new("make")
-                .current_dir(&rust_build_path)
-                .arg("-j6")
-                .output()
-                .expect("Fail to build unicorn on *nix.");
-
-            println!(
-                "cargo:rustc-link-search={}",
-                rust_build_path.to_str().unwrap()
-            );
-        }
-    }
+    let manifest_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let manifest_path = Path::new(&manifest_dir).to_path_buf();
+    let src_dir = manifest_path.parent().unwrap().parent().unwrap();
+
+    #[cfg(target_os = "windows")]
+    setup_env_windows();
+
+    // need to clear build target and append "build" to the path because
+    // unicorn's CMakeLists.txt doesn't properly support 'install', so we use
+    // the build artifacts from the build directory, which cmake crate sets
+    // to "<out_dir>/build/"
+    let dst = cmake::Config::new(src_dir)
```

Comment:
You seem to forget Debug/Release switch.

Reply:
it's handled by cmake crate